### PR TITLE
Init SpeedRunFetcher.ex

### PR DESCRIPTION
**Pull request by kenspirit**

### CODE REVIEW

File: snippets/SpeedRunFetcher.ex [hunk 1]
```diff
@@ -0,0 +1,35 @@
+defmodule CodesOnChain.SpeedRunFetcher do
+  alias Components.ExHttp
+
+  def fetch_data(addr, speedrun_api_endpoint, speedrun_url) do
+    {:ok, data} = ExHttp.http_get(speedrun_api_endpoint)
```
Comment (leeduckgo) [inline snippets/SpeedRunFetcher.ex:5]: "#{speedrun_api_endpoint}/builders/"

File: snippets/SpeedRunFetcher.ex [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    runner =
+      data
+      |> Enum.find(fn %{"id" => id} ->
+        id == addr
+      end)
+
+    role = Map.get(runner, "role")
+
+    accepted_challengues =
```
Comment (leeduckgo) [inline snippets/SpeedRunFetcher.ex:15]: typo

File: snippets/SpeedRunFetcher.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    len = length(accepted_challengues)
+
+    %{
+      chanllege_passed_num: len,
+      link: "#{speedrun_url}#{role}s/#{addr}",
+      level: if(len == 0, do: 0, else: fib(len))
+    }
+  end
+
+  defp fib(0), do: 1
```
Comment (leeduckgo) [inline snippets/SpeedRunFetcher.ex:32]: why fib?🤣

File: snippets/SpeedRunFetcher.ex [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      |> Map.to_list()
+      |> Enum.filter(fn {_name, challenge} ->
+        Map.get(challenge, "status") == "ACCEPTED"
+      end)
+
+    len = length(accepted_challengues)
+
+    %{
+      chanllege_passed_num: len,
+      link: "#{speedrun_url}#{role}s/#{addr}",
```
Comment (leeduckgo) [inline snippets/SpeedRunFetcher.ex:27]: just "builders" is ok

File: snippets/SpeedRunFetcher.ex (hunk 3 above)
Comment (kenspirit) [reply snippets/SpeedRunFetcher.ex:32]: > why fib?🤣

就先随便搞的。Level 是数字也不知道能有啥。和 num 一样没啥意思，就先弄个 fib 了，反正级别也不会多吧？

**Pull request merged**